jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/PerProcessInstanceRuntimeManager.java

```diff
@@ -162,12 +162,17 @@ public RuntimeEngine getRuntimeEngine(Context<?> context) {
     public void signalEvent(String type, Object event) {
         
         // first signal with new context in case there are start event with signal
+        KieSession signalSession = null;
         RuntimeEngine runtimeEngine = getRuntimeEngine(ProcessInstanceIdContext.get());
         try {
             // signal execution can rise errors
-            runtimeEngine.getKieSession().signalEvent(type, event);
+            signalSession = runtimeEngine.getKieSession();
+            signalSession.signalEvent(type, event);
         } finally {
             // ensure we clean up
+            if(signalSession!=null) {
+                signalSession.execute(new DestroyKSessionCommand(signalSession, this));
```

Comment:
not very sure this is valid as there is no tx during this causing the tx sync not to be called.

Reply:
ok just tested the right approach but it would require a huge amount of work to fix it properly. this will work.